tasks/resolve_reference.rb

```diff
@@ -56,6 +72,31 @@ def resolve_reference(opts)
   end
 
   # private
+  def ad_entry_filter_dns_hostnames(entry, filter_dns_hostnames)
+    return false unless entry.attribute_names.include?(:dnshostname)
+    return false if entry.dnshostname.nil? || entry.dnshostname.empty?
+    entry_dns_hostname = entry.dnshostname[0]
+    filter_dns_hostnames.include?(entry_dns_hostname)
+  end
+
+  def ldap_to_unix_time(ldap_time_s)
```

Comment:
By filtering at the server level, we can avoid all of the timestamp conversion issues.

Reply:
Sort of, but not really.. We still need to ask the LDAP server to based on a timestamp in LDAP time, so we have to convert from `now - 30 days` and then convert that to LDAP time as a reference.